# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/windows/nsClipboard.h
# Commit: a4b80c7849fe
# Full Hash: a4b80c7849febd8d072b4cebfa30606f849877c8
# Author: mspiess <marco.spiess@hotmail.de>
# Date: 2023-03-22 09:41:36
# Regressor Bug: 580928
# File Overlap Count: 2
# Description:
#   Bug 580928 - Add support for Drag and Drop with Outlook items. r=cmartin
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D157587
# ==============================================================================

diff -r d65395a2bd80 -r a4b80c7849fe widget/windows/nsClipboard.h
--- a/widget/windows/nsClipboard.h	Tue Mar 21 16:31:56 2023 +0200
+++ b/widget/windows/nsClipboard.h	Tue Mar 21 13:50:32 2023 +0000
@@ -66,6 +66,8 @@
   // of Gecko.
   static UINT GetFormat(const char* aMimeStr, bool aMapHTMLMime = true);
 
+  static UINT GetClipboardFileDescriptorFormatA();
+  static UINT GetClipboardFileDescriptorFormatW();
   static UINT GetHtmlClipboardFormat();
   static UINT GetCustomClipboardFormat();
 
@@ -90,6 +92,12 @@
                                uint32_t* outDataLen);
 
   static void ResolveShortcut(nsIFile* inFileName, nsACString& outURL);
+  static nsresult GetTempFilePath(const nsAString& aFileName,
+                                  nsAString& aFilePath);
+  static nsresult SaveIStorage(IDataObject* aDataObject, UINT aIndex,
+                               const nsAString& aFileName);
+  static nsresult SaveIStream(IDataObject* aDataObject, UINT aIndex,
+                              const nsString& aFileName);
 
   nsIWidget* mWindow;
 };